paseos/power/is_in_eclipse.py

```diff
@@ -0,0 +1,86 @@
+"""This file provides method for computing whether an actor is currently in eclipse"""
+import pykep as pk
+import numpy as np
+from skspatial.objects import LineSegment, Line, Sphere
+from loguru import logger
+
+from paseos.actors.base_actor import BaseActor
+
+
+def is_in_eclipse(
+    actor: BaseActor, central_body: pk.planet, t: pk.epoch, plot=False
```

Comment:
Should we at least raise a warning if the actor's central body is  pk.planet.jpl_lp("earth")?

Reply:
The code works with any central body assuming the central body is described in the reference frame of its star, not sure how to check for that 🤔

Reply:
Ok, maybe I am confused by the nomenclature (heliocentric) and by this:

 ```line_between_sun_and_actor = Line(
        [0, 0, 0],
        r_sat_heliocentric,
    )```

I thought this assumed [0,0,0] to be the position of the sun.
    
    I

Reply:
Yes but the sun (or other star) is always at 0,0,0 in heliocentric frame, that is the definition of it